esp-hal/src/gpio/mod.rs

```diff
@@ -2250,37 +2255,36 @@ impl RtcPin for AnyPin {
     }
 
     #[cfg(any(xtensa, esp32c6))]
-    fn rtc_set_config(&mut self, input_enable: bool, mux: bool, func: RtcFunction) {
+    fn rtc_set_config(&self, input_enable: bool, mux: bool, func: RtcFunction) {
         handle_rtcio!(self, target, {
-            RtcPin::rtc_set_config(&mut target, input_enable, mux, func)
+            RtcPin::rtc_set_config(&target, input_enable, mux, func)
         })
     }
 
-    fn rtcio_pad_hold(&mut self, enable: bool) {
-        handle_rtcio!(self, target, {
-            RtcPin::rtcio_pad_hold(&mut target, enable)
-        })
+    #[allow(unused_braces, reason = "False positive")]
```

Comment:
Curious, do you remember what braces were failing? The method is very similar to `rtc_set_config` which doesn't seem to complain.

Reply:
```
warning: unnecessary braces around block return value
    --> C:\_Espressif\esp-hal\esp-hal\src\gpio\mod.rs:2265:37
     |
2265 |         handle_rtcio!(self, target, { RtcPin::rtcio_pad_hold(&target, enable) })
     |                                     ^^                                       ^^
     |
     = note: `#[warn(unused_braces)]` on by default
help: remove these braces
     |
2265 -         handle_rtcio!(self, target, { RtcPin::rtcio_pad_hold(&target, enable) })
2265 +         handle_rtcio!(self, target, RtcPin::rtcio_pad_hold(&target, enable))
```

This is some weird interaction between using a macro, and the whole expression being in a single line. If `rtc_set_config` was a bit shorter, it would trigger the same warning.

Reply:
https://github.com/rust-lang/rust/issues/88104